src/main/java/com/google/sps/servlets/GoServlet.java

```diff
@@ -61,33 +73,78 @@ public GoServlet() throws IOException {
    */
   public GoServlet(
       DirectionsClientFactory directionsClientFactory,
+      TasksClientFactory tasksClientFactory,
       String fakeApiKey,
       String fakeOrigin,
       String fakeDestination,
       List<String> fakeWaypoints) {
     this.directionsClientFactory = directionsClientFactory;
+    this.tasksClientFactory = tasksClientFactory;
     apiKey = fakeApiKey;
     origin = fakeOrigin;
     destination = fakeDestination;
     waypoints = fakeWaypoints;
   }
 
+  private List<Task> getTasks(TasksClient tasksClient) throws IOException {
+    List<TaskList> taskLists = tasksClient.listTaskLists();
+    List<Task> tasks = new ArrayList<>();
+    for (TaskList taskList : taskLists) {
+      tasks.addAll(tasksClient.listTasks(taskList));
+    }
+    return tasks;
+  }
+
+  /**
+   * Parses for locations in Tasks. Scope is public for testing purposes.
+   *
+   * @param prefix String which represents the prefix wrapped in square brackets to look for. (e.g.
+   *     Location if looking for [Location: ])
+   * @param tasks List of tasks to parse for locations from.
+   * @return List of strings representing the locations.
+   */
+  public List<String> getLocations(String prefix, List<Task> tasks) {
```

Comment:
If this logic is separate enough to be tested in isolation as a public method, it should probably be in a client as a static method.

Reply:
I agree that it should probably be somewhere else, I have moved it into a Locations Utility class since I don't think it fits into any of the existing clients and that it requires any dependency injection like the other clients we have. Thanks for pointing that out!

Reply:
`RESOLVED`